src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs

```diff
@@ -150,7 +153,19 @@ private bool CanMoveDown(int param)
 
         private bool CanUpdate(int param)
         {
-            return RootLocations.IndexOf(Resources.PackagePathViewModel_Standard_Library) != param;
+            var programDataPackagePathIndex = GetIndexOfProgramDataPackagePath();
+
+            //editing builtin packages or programData package paths is not allowed.
+            return RootLocations.IndexOf(Resources.PackagePathViewModel_Standard_Library) != param &&
+                programDataPackagePathIndex != param;
+        }
+
+        private int GetIndexOfProgramDataPackagePath()
+        {
+            var programDataPath = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
+            var programDataPackagePath = RootLocations.Where(x => x.StartsWith(programDataPath)).FirstOrDefault();
+            var programDataPackagePathIndex = RootLocations.IndexOf(programDataPackagePath);
+            return programDataPackagePathIndex;
```

Comment:
We are only tracking the first program data directory found and I guess a user could add an additional program data location themselves? That could mess things up big time (even without the changes in this PR) but is not very likely to happen I think. And we are hopefully removing program data directory soon.

Reply:
hey @sm6srw are you suggesting this method should return a list of ints?